### DISCUSSION TITLE
Automatic revisions

### DISCUSSION THREAD

**Discussion by amitaibu** [digitallyinduced/ihp#1349] [Q&A]
It's possible to have Postgress help with the creation of [revisions](https://stackoverflow.com/a/2259527/750039) of records with triggers (e.g. whenever a blog post's title changes, save it as a revision).

I'm wondering if there's a way to do that with triggers, that would also capture the "current user". For example, when creating revisions in Drupal (isn't done with triggers), one can see when and who created that revision.

![image](https://user-images.githubusercontent.com/125707/152683560-ccfc5bc0-349e-4de7-87be-be2763ae5c0b.png)

**Comment by mpscholten** [comment]
That seems possible. The triggers can capture the current user using `ihp_user_id()` if set up the right way 👍

**Reply by amitaibu** [reply]
I tried to follow the [Stackoverflow](https://stackoverflow.com/questions/2259158/track-revisions-in-postgresql/2259527#2259527) answer, trying to setup a `revision_posts` table for `posts`.

1) If I try to add the following to `Fixtures.sql`

`CREATE TABLE revision_posts ( revision SERIAL NOT NULL) INHERITS (posts);`

we get an error

![image](https://user-images.githubusercontent.com/125707/152697678-b0bc803a-1724-4d39-a2c9-952706e4bb71.png)

2) Once we'll figure (1), should the trigger functions should be added to `Schema.sql`?

3) How would you use the `ihp_user_id`? I assume it's somewhere inside `INSERT INTO revision_posts SELECT NEW.*;`?
